File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.js
function solution(board)
{
    const [R, C] = [board.length, board[0].length];
    
    for(let i = 1; i < R; i++) {
        for(let j = 1; j < C; j++) {
            if(board[i][j] === 1) {
                const upLeft = board[i - 1][j - 1];
                const left = board[i][j - 1];
                const up = board[i - 1][j];
                
                board[i][j] = Math.min(upLeft, left, up) + 1;
            }
        }
    }
    
    const sqrt = Math.max(...board.map(row => Math.max(...row)));
    return Math.pow(sqrt, 2);
}

File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.py
def solution(board):
    n, m = len(board), len(board[0])
    for i in range(1, n):
        for j in range(1, m):
            if board[i][j] == 1:
                board[i][j] = min(
                    board[i - 1][j], board[i][j - 1], board[i - 1][j - 1]
                ) + 1
            
    return max(max(row) for row in board) ** 2